plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -341,7 +343,7 @@ public static Object deserializePartitionValue(Type type, String valueString, St
                 String partitionValue;
                 if (type.typeId() == FIXED || type.typeId() == BINARY) {
                     // this is safe because Iceberg PartitionData directly wraps the byte array
-                    partitionValue = new String(((ByteBuffer) value).array(), UTF_8);
+                    partitionValue = StandardCharsets.ISO_8859_1.decode(Base64.getEncoder().encode((ByteBuffer) value)).toString();
```

Comment:
1. use `Base64.getEncoder().encodeToString` instead of `StandardCharsets.ISO_8859_1.decode`
2. does the comment in the preceding line require an update?

Reply:
1. sure
2. The comment seems to be valid.